daemon/cmd/daemon_main.go

```diff
@@ -1867,8 +1867,9 @@ func initKubeProxyReplacementOptions() {
 			_, found2 = h["bpf_get_netns_cookie"]
 		}
 		if !(found1 && found2) {
-			log.Warnf("sessionAffinity for host reachable services needs kernel 5.7.0 or newer. " +
-				"Disabling sessionAffinity for cases when a service is accessed from a cluster.")
+			log.Warnf("sessionAffinity for host reachable services needs kernel 5.7.0 or newer " +
+				"to work properly: the same service endpoint will be selected from all network " +
+				"namespaces on the host.")
```

Comment:
Looks good overall. I think here this needs a bit of rewrite. It may not be clear to the user that this is for E-W traffic-only but not for N-S one.

Reply:
Another idea to improve this for older kernels:

- Given ClusterIP is non-routable, we leave this as-is what you have here for ClusterIP
- For everything else we route the traffic to the front-end node, meaning E-W traffic would be service-processed similar way as N-S traffic through bpf_host NodePort LB

For the latter, this means that in the service map for services with sessionAffinity we reroute e.g. 127.0.0.1:NodePort to the front-end instead of backend.

Wdyt?